test/api/capi/test_capi_complex_types.cpp

```diff
@@ -757,3 +756,83 @@ TEST_CASE("Array value construction") {
 	}
 	duckdb_destroy_value(&array_value);
 }
+
+TEST_CASE("Map value construction") {
+	auto key_type = duckdb_create_logical_type(DUCKDB_TYPE_INTEGER);
+	auto value_type = duckdb_create_logical_type(DUCKDB_TYPE_DOUBLE);
+	auto map_type = duckdb_create_map_type(key_type, value_type);
+
+	duckdb::vector<duckdb_value> keys;
+	keys.push_back(duckdb_create_int64(42));
+	keys.push_back(duckdb_create_int64(43));
+	keys.push_back(duckdb_create_int64(44));
+
+	duckdb::vector<duckdb_value> values;
+	values.push_back(duckdb_create_double(42.0));
+	values.push_back(duckdb_create_double(43.0));
+	values.push_back(duckdb_create_double(44.0));
+
+	duckdb_value keys_v = duckdb_create_list_value(key_type, keys.data(), keys.size());
+	duckdb_value values_v = duckdb_create_list_value(value_type, values.data(), values.size());
+
+	auto map_value = duckdb_create_map_value(map_type, keys_v, values_v);
+	REQUIRE(map_value);
+
+	duckdb_destroy_value(&keys_v);
+	duckdb_destroy_value(&values_v);
+
+	REQUIRE(duckdb_get_map_size(map_value) == 3);
+
+	for (idx_t i = 0; i < 3; i++) {
+		auto key = duckdb_get_map_key(map_value, i);
+		REQUIRE(duckdb_get_int64(key) == 42L + (int64_t)i);
+		duckdb_destroy_value(&key);
+
+		auto value = duckdb_get_map_value(map_value, i);
+		REQUIRE(duckdb_get_double(value) == 42.0 + i);
+		duckdb_destroy_value(&value);
+	}
+
+	duckdb_destroy_logical_type(&key_type);
+	duckdb_destroy_logical_type(&value_type);
+	duckdb_destroy_logical_type(&map_type);
+
+	for (auto &val : keys) {
+		duckdb_destroy_value(&val);
+	}
+
+	for (auto &val : values) {
+		duckdb_destroy_value(&val);
+	}
+
+	duckdb_destroy_value(&map_value);
+}
+
+TEST_CASE("Union value construction") {
```

Comment:
This could also benefit from some more unhappy path testing:
- `union_type` as not a UNION type
- `tag` as a value bigger than or equal to the number of members in the union